### ISSUE TITLE
error with fused_bias_act_kernel.cu

### ISSUE THREAD

**Issue by tankche1** [wpeebles/gangealing#19]
Hi,
I try to run
```
CUDA_VISIBLE_DEVICES=7 python train.py  --ckpt cat --load_G_only --padding_mode border --vis_every 5000 --ckpt_every 50000  --iter 1500000 --tv_weight 2500 --loss_fn lpips --exp-name lsun_cats_lpips
```
and get several different errors when compiling fused_bias_act_kernel.cu according to the PyTorch and Cuda version I use. I have try pytorch 1.2.1, 1.4.0, 1.8.0 with cuda 10.0, 10.1 , 11.1.

Here is an example error with pytorch 1.8.0, cuda 11.1:

```
Traceback (most recent call last):
  File "train.py", line 13, in <module>
    from models import Generator, get_stn, DirectionInterpolator, PCA
  File "/home/username/nvidia_code/gangealing/models/__init__.py", line 2, in <module>
    from models.stylegan2.networks import Generator
  File "/home/username/nvidia_code/gangealing/models/stylegan2/networks.py", line 7, in <module>
    from models.stylegan2.op import FusedLeakyReLU, fused_leaky_relu, upfirdn2d, conv2d_gradfix
  File "/home/username/nvidia_code/gangealing/models/stylegan2/op/__init__.py", line 1, in <module>
    from .fused_act import FusedLeakyReLU, fused_leaky_relu
  File "/home/username/nvidia_code/gangealing/models/stylegan2/op/fused_act.py", line 15, in <module>
    os.path.join(module_path, "fused_bias_act_kernel.cu"),
  File "/home/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch/utils/cpp_extension.py", line 1091, in load
    keep_intermediates=keep_intermediates)
  File "/home/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch/utils/cpp_extension.py", line 1302, in _jit_compile
    is_standalone=is_standalone)
  File "/home/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch/utils/cpp_extension.py", line 1407, in _write_ninja_file_and_build_library
    error_prefix=f"Error building extension '{name}'")
  File "/home/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch/utils/cpp_extension.py", line 1683, in _run_ninja_build
    raise RuntimeError(message) from e
RuntimeError: Error building extension 'fused': [1/2] /cm/shared/apps/cuda100/10.0.130/bin/nvcc --generate-dependencies-with-compile --dependency-output fused_bias_act_kernel.cuda.o.d -D
TORCH_EXTENSION_NAME=fused -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -isystem /home/zitianch
en/anaconda2/envs/gg/lib/python3.7/site-packages/torch/include -isystem /home/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -isystem /home
/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch/include/TH -isystem /home/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch/include/THC -isystem /cm/shared/a
pps/cuda100/10.0.130/include -isystem /home/username/anaconda2/envs/gg/include/python3.7m -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUD
A_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_75,code=compute_75 -gencode=arch=compute_75,code=sm_75 --compiler-options '-fPIC'
 -std=c++14 -c /home/username/nvidia_code/gangealing/models/stylegan2/op/fused_bias_act_kernel.cu -o fused_bias_act_kernel.cuda.o
FAILED: fused_bias_act_kernel.cuda.o
/cm/shared/apps/cuda100/10.0.130/bin/nvcc --generate-dependencies-with-compile --dependency-output fused_bias_act_kernel.cuda.o.d -DTORCH_EXTENSION_NAME=fused -DTORCH_API_INCLUDE_EXTENSI
ON_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -isystem /home/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch
/include -isystem /home/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -isystem /home/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch/include/TH -isystem /home/username/anaconda2/envs/gg/lib/python3.7/site-packages/torch/include/THC -isystem /cm/shared/apps/cuda100/10.0.130/include -isystem /home/username
/anaconda2/envs/gg/include/python3.7m -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATOR
S__ --expt-relaxed-constexpr -gencode=arch=compute_75,code=compute_75 -gencode=arch=compute_75,code=sm_75 --compiler-options '-fPIC' -std=c++14 -c /home/username/nvidia_code/gangealing
/models/stylegan2/op/fused_bias_act_kernel.cu -o fused_bias_act_kernel.cuda.o
nvcc fatal   : Unknown option '-generate-dependencies-with-compile'
ninja: build stopped: subcommand failed.
```
Any clue to solve this? I already check the stylegan2 repository but found nothing. Can you tell me what pytorch and cuda version I should use?

**Comment by wpeebles** [comment]
Hi @tankche1, I've been able to train with both CUDA 10.2 and 11.3, and you'll want to use a recent version of PyTorch (1.10.1+). Can you try using one of our included conda environments,  either `environment.yml` or `environment_cu102.yml`? Example usage to create the conda environment: `conda env create -f environment.yml`

If you continue to see errors with our environments, it's possible there may be a problem with your CUDA installation. For example, if `nvcc -V` produces an error, CUDA may need to be reinstalled or you may need to change some environment variables like `LD_LIBRARY_PATH`.

By the way, unrelated to this issue, but when training with 1 gpu you'll need to increase the per-gpu batch size with the `--batch` argument when calling `train.py` in order to get high quality results. Ideally, total batch size over all GPUs should be 40, so with 1 gpu you might want to try `--batch 40`.

**Comment by tankche1** [comment]
Problem solved by using pytorch 1.11.0 and cuda 10.0.
